common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -960,14 +960,26 @@ public static String toSocketAddressString(InetSocketAddress addr) {
         return sb.append(':').append(port).toString();
     }
 
+    /**
+     * Returns the {@link String} representation of a host port combo.
+     */
+    public static String toSocketAddressString(String hostString, int port) {
+        return newSocketAddressStringBuilder(
+                hostString, String.valueOf(port), isValidIpV4Address(hostString)).append(port).toString();
+    }
+
     private static StringBuilder newSocketAddressStringBuilder(String hostString, String port, boolean ipv4) {
         if (ipv4) {
             // Need to include enough space for hostString:port.
             return new StringBuilder(hostString.length() + 1 + port.length()).append(hostString);
         }
         // Need to include enough space for [hostString]:port.
-        return new StringBuilder(
-                hostString.length() + 3 + port.length()).append('[').append(hostString).append(']');
+        StringBuilder stringBuilder = new StringBuilder(
+                hostString.length() + 3 + port.length());
+        if (hostString.startsWith("[") &&  hostString.endsWith("]")) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This call to String.startsWith can be rewritten using String.charAt(0) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifyStartsWith)